src/Microsoft.Health.Dicom.Functions.UnitTests/IndexMetricsCollection/IndexMetricsCollectionFunctionTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Features.Telemetry;
+using Microsoft.Health.Dicom.Functions.IndexMetricsCollection;
+using Microsoft.Health.Dicom.Functions.IndexMetricsCollection.Telemetry;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Functions.UnitTests.IndexMetricsCollection;
+
+public class IndexMetricsCollectionFunctionTests
+{
+    private readonly IndexMetricsCollectionFunction _collectionFunction;
+    private readonly IIndexDataStore _indexStore;
+    private readonly IndexMetricsCollectionMeter _meter;
+    private List<Metric> _exportedItems;
+    private MeterProvider _meterProvider;
+    private readonly TimerInfo _timer;
+
+    public IndexMetricsCollectionFunctionTests()
+    {
+        _meter = new IndexMetricsCollectionMeter();
+        _indexStore = Substitute.For<IIndexDataStore>();
+        _collectionFunction = new IndexMetricsCollectionFunction(
+            _indexStore,
+            Options.Create(new FeatureConfiguration { EnableExternalStore = true, }),
+            _meter);
+        _timer = Substitute.For<TimerInfo>(default, default, default);
+    }
+
+    private void InitializeMetricExporter()
+    {
+        _exportedItems = new List<Metric>();
+        _meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddMeter($"{OpenTelemetryLabels.BaseMeterName}.{IndexMetricsCollectionMeter.MeterName}")
```

Comment:
I would create an `internal` ctor for `IndexMetricsCollectionMeter` that allows users to specify a name. Then randomly generate a name per test. Otherwise, observationally, there is a risk that metrics will be collected by other in-memory exporters.

E.g.
```csharp
string meterName = Guid.NewGuid().ToString();
_meter = new IndexMetricsCollectionMeter(meterName);
_meterProvider = Sdk
    .CreateMeterProviderBuilder()
    .AddMeter(meterName)
    .AddInMemoryExporter(...)
    .Build();
```